addons/mail/models/discuss/discuss_channel.py

```diff
@@ -160,6 +161,17 @@ def _constraint_group_id_channel(self):
 
     # COMPUTE / INVERSE
 
+    @api.depends("channel_member_ids", "auto_recompute_name")
+    def _compute_name(self):
+        for channel in self:
+            if channel.auto_recompute_name:
+                partners_to_name = channel.channel_member_ids.partner_id[:5]
+                new_name = partners_to_name[0].name if partners_to_name else ""
+                if len(partners_to_name) > 1:
+                    new_name = ', '.join(partners_to_name[0:-1].mapped('name')) + ' and ' + partners_to_name[-1].name
```

Comment:
should use `format_list` probably, but this is a huge drawback to do it in python: it will be computed in the current language of the current user.

Reply:
I prefer to keep it like this so that we don't get bothered by the lang problem. The first channel member name you mentioned above is the part of the computing -> if there's only one member, use his/her name, otherwise we compute it with the `and`.

Reply:
Do we _**really**_ need the leading "and"?
I think we make this a huge problem because we don't allow us to relax the format a bit to look to be language-agnostic with only "," joins.

Reply:
Note that even our competitors don't put the "and" at the end, and it's actually better because you see all names better and less characters truncated the better (keeping ", " for visual separation)

Reply:
Using `,` as separator is not something that works in all languages either.

But I guess it's still better than using "and" and other words.